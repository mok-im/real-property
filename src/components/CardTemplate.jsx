import React from "react";
import { useHistory } from "react-router-dom";
import { Card, ListGroup } from "react-bootstrap";
import styled from "styled-components";

const CardHolder = styled.div`
  transition: all 500ms ease;
  &:hover {
    box-shadow: 0px 12px 30px 1px rgba(34, 60, 80, 0.25);
    cursor: pointer;
    transform: translate(0, -5px);
  }
`;

const CardTemplate = ({ card }) => {
  const history = useHistory();

  return (
    <CardHolder>
      <Card onClick={() => history.push("/house/" + card.id)} className="">
        <img
          fluid
          style={{ height: "350px" }}
          src={card.img}
          className="d-block w-100"
          alt={card.id}
        />
        <Card.Body>
          <Card.Title>{card.name_house}</Card.Title>
          <Card.Text>{card.description}</Card.Text>
          <ListGroup.Item>Price: {card.price}</ListGroup.Item>
          <ListGroup.Item>Address: {card.address}</ListGroup.Item>
        </Card.Body>
      </Card>
    </CardHolder>
  );
};

export default CardTemplate;
