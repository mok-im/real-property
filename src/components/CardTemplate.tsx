import { FC } from 'react';
import { useHistory } from 'react-router-dom';

import { Card, ListGroup } from 'react-bootstrap';

import { CardHolder } from '../styled/CardHolder';
import { CardImg } from '../styled/CardImg-styled';
import { CardProps } from '../types';

const CardTemplate: FC<CardProps> = ({ card }) => {
    const history = useHistory();

    return (
        <CardHolder>
            <Card onClick={() => history.push('/house/' + card.id)}>
                <CardImg src={card.img} alt="house" />
                <Card.Body>
                    <Card.Title>{card.name_house}</Card.Title>
                    <Card.Text>{card.description}</Card.Text>
                    <ListGroup.Item>Price: {card.price}</ListGroup.Item>
                    <ListGroup.Item>Address: {card.address}</ListGroup.Item>
                </Card.Body>
            </Card>
        </CardHolder>
    );
};

export default CardTemplate;
