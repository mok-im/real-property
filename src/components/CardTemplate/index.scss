.cardholder {
  transition: all 500ms ease;
  &:hover {
    box-shadow: 0px 1px 25px 5px rgba(63, 85, 172, 0.267);
    cursor: pointer;
    transform: translate(0, -5px);
  }
}
