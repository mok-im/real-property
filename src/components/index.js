import CardTemplate from './CardTemplate'
export default CardTemplate

