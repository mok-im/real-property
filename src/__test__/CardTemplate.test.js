import React from 'react';
import { shallow } from 'enzyme';
import { CardTemplate } from '../components';
import toJson from 'enzyme-to-json';

const props = {
    card: {
        id: 1,
        name_house: "House 1",
        price: "1000$",
        address: "New York City1",
        img: "https://ap.rdcpix.com/ce4305635b3d44aef2e6abe0ce8f1f93l-m507177317od-w1024_h768_x2.webp",
        description: "Lorem1 ",
        sales_name: "John1",
    }
};

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    })
}));

describe('Testing component <CardTemplate />', () => {
    it('should render Card template', () => {
        const component = shallow(<CardTemplate {...props} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Test click event', () => {
        const button = shallow((<CardTemplate {...props} onClick={mockHistoryPush} />));
        button.find('Card').simulate('click');
        expect(mockHistoryPush.mock.calls.length).toEqual(1);
    });
});
