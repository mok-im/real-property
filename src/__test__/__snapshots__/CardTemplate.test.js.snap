// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <CardTemplate /> should render Card template 1`] = `
<styled.div>
  <Card
    body={false}
    onClick={[Function]}
  >
    <styled.img
      alt="house"
      src="https://ap.rdcpix.com/ce4305635b3d44aef2e6abe0ce8f1f93l-m507177317od-w1024_h768_x2.webp"
    />
    <CardBody>
      <CardTitle>
        House 1
      </CardTitle>
      <CardText>
        Lorem1 
      </CardText>
      <ListGroupItem
        active={false}
        disabled={false}
      >
        Price: 
        1000$
      </ListGroupItem>
      <ListGroupItem
        active={false}
        disabled={false}
      >
        Address: 
        New York City1
      </ListGroupItem>
    </CardBody>
  </Card>
</styled.div>
`;
